cyten/models/couplings.py

```diff
@@ -0,0 +1,757 @@
+            factorization will have that dtype. By default, we detect the dtype from the block.
+        """
+        backend = get_same_DOF_backend(*sites)
+        device = get_same_DOF_device(*sites)
+        co_domain = [s.leg for s in sites]
+        p_labels = [f'p{i}' for i in range(len(sites))]
+        labels = [*p_labels, *[f'{pi}*' for pi in p_labels][::-1]]
+        op = SymmetricTensor.from_dense_block(operator, co_domain, co_domain, backend=backend,
+                                              labels=labels, dtype=dtype, device=device,
+                                              understood_braiding=understood_braiding)
+        return cls.from_tensor(op, sites=sites, name=name)
+
+    @classmethod
+    def from_tensor(cls, operator: SymmetricTensor, sites: list[DegreeOfFreedom], name: str = None
+                    ) -> Coupling | OnSiteOperator:
+        """Convert an operator / tensor to a :class:Coupling.
+        
+        Decompose a (in general multi-site) operator into factors using SVD such that contracting
+        the factors again reproduces the operator.
+
+        .. note ::
+            For symmetries with non-symmetric braids, the decomposition depends on the levels of
+            the legs that determine whether over-braids or under-braids occur when exchanging legs.
+            The convention here is to assign higher levels to legs "further to the right", i.e.,
+            the legs corresponding to the labels `p2` and `p2*` have higher levels than `p1` and
+            `p1*`, and lower levels than `p3` and `p3*`.
+
+        Parameters
+        ----------
+        operator : :class:`SymmetricTensor`
+            Operator to be converted to a coupling. The legs should be ordered as
+            ``[p0, p1, ..., p1*, p0*]``, where ``pi`` and ``pi*`` correspond to the legs associated
+            with site ``sites[i]``.
+        sites : list of :class:`Site`
+            The sites that the operator acts on.
+        name : str, optional
+            A descriptive name that can be used when pretty-printing, to identify the coupling.
+            For example, a Heisenberg coupling is usually initialized with name ``'S.S'``.
+        """
+        assert operator.backend == get_same_DOF_backend(*sites)
+        assert operator.device == get_same_DOF_device(*sites)
+        assert operator.codomain.factors == [site.leg for site in sites]
+        assert operator.domain.factors == operator.codomain.factors
+
+        if len(sites) == 1:
+            return OnSiteOperator.from_tensor(operator, sites, name=name)
+
+        # decompose from right to left using SVD
+        # convention for levels: legs to the right have higher levels
+        factorization = []
+        n = operator.num_codomain_legs
+        levels = [2 * i for i in range(n)]
+        levels.extend([2 * i + 1 for i in range(n)][::-1])
+
+        U = permute_legs(operator, domain=[n, n - 1], levels=levels, bend_right=True)
```

Comment:
Ok, i see why we need levels here.
I assume that the choice does not affect the outcome, since we braid back after the SVD?
If yes, remove the note in the docstring and explain here.
If no, I am surprised and need to think about this more.

Maybe we want to offer this type of "horizontal splitting" as a separate function?


```
   |  |  |  |       |   |  |  |
   ##########   =   #---#######
   |  |  |  |       |   |  |  |
```

Reply:
I think this resolves nicely if we use left bends where needed, then we can do this braid-free